### PR TITLE
Further optimize blockchain transaction inserts to DB

### PR DESCRIPTION
**Pull request by peterbroadhurst**
In PR chain with #1343 

- Creates a new `txwriter` package that depends on both `operations` and `txcommon` to do batched inserts
   - Dispatches to a pool of background workers
   - Assigns a new ID to each transaction in the batch
   - Splits to transactions with/without `idempotencyKey` set
   - Attempts to insert all with new `InsertTransactions` multi-insert function in DB layer
   - Handles partial results on the `idempotencyKey` set TX, by querying those idempotency keys
   - Compares ID of the transactions in the DB, with the inserts, to find the duplicates
   - Inserts the `operations` with a new multi-insert function in the DB layer (these always have new IDs)
- Updates `contracts` package to use ^^^ instead of one-by-one insertion of tx+ops
   - The inbound thread for invoke/deploy requests doesn't even have a DB transaction now (`RunAsGroup` call removed)
   - Creation of the `core.Operation` moved up the function, as need to be passed in
   - Idempotency handling to resubmit operations in the case of a clash, unchanged
- Updates blockchain connector interface to split parsing with execution
   - We were compiling the FFI schema twice per transaction in the blockchain connector
- Adds caching to `contracts` package
   - For requests referencing an FFI: FFI ID + `methodPath` -> `FFIMethod` + `[]*FFIError`
   - For all requests: Hash of compiled JSON Schema, to the JSON Schema object
   - Also builds a hash-of-hashes across the combination of method+errors, which is used to cache the result from the blockchain connector validation

### CODE REVIEW

**Comment by nguyer** [comment]
@peterbroadhurst It does look like we have a coverage miss here: https://app.codecov.io/gh/hyperledger/firefly/pull/1354/blob/internal/operations/context.go

File: internal/contracts/manager.go [hunk 1]
```diff
@@ -374,39 +400,34 @@ func (cm *contractManager) InvokeContract(ctx context.Context, req *core.Contrac
...
+	}
+	var op *core.Operation
+	if req.Type == core.CallTypeInvoke {
+		op, err = cm.writeInvokeTransaction(ctx, req)
+		if err != nil {
+			if _, ok := err.(*sqlcommon.IdempotencyError); ok {
+				if op != nil {
+					// Idempotency key clash but we resubmitted an initialized operation? Return 20x, not 409
+					return op, nil
+				}
```
[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/contracts/manager.go:426]: Great question... digging

[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/contracts/manager.go:426]: ok - I've worked through the code, and did find (existing I think in all cases) issues in how the TXID was being propagated, and a weirdness on an interface that returned a single operation even though there might be more.

Commit added

[the remark this answers is not in the document]
Comment (peterbroadhurst) [reply internal/contracts/manager.go]: https://github.com/hyperledger/firefly/pull/1354/commits/e40ec89faa376ad9c35962597ee57be83eff41d2

**Review by nguyer — APPROVED** [review]

**Pull request merged**